packages/core/src/modules/connections/ConnectionsApi.ts

```diff
@@ -227,6 +230,42 @@ export class ConnectionsApi {
     return connectionRecord
   }
 
+
+  /**
+   * Send a trust ping to an established connection
+   *
+   * @param connectionId the id of the connection for which to accept the response
+   * @param responseRequested do we want a response to our ping
+   * @param withReturnRouting do we want a response at the time of posting
+   * @returns boolean indicating success or failure
+   */
+  public async trustPing(connectionId: string, responseRequested=true, withReturnRouting=false): Promise<boolean> {
+    const connectionRecord = await this.connectionService.getById(this.agentContext, connectionId)
+
+    let outboundMessageContext
+    if (connectionRecord.state === DidExchangeState.Completed) {
```

Comment:
The createTrustPing message check the state of the connection record and asserts it is ResponseReceived or Completed. We can leave this check here. 

That way we also don't have to return true / false

Reply:
Fixed